test-common/src/main/scala/org/scalajs/testcommon/Serializer.scala

```diff
@@ -0,0 +1,290 @@
+package org.scalajs.testcommon
+
+import sbt.testing._
+import java.io._
+
+private[scalajs] trait Serializer[T] {
+  def serialize(x: T, out: Serializer.SerializeState): Unit
+  def deserialize(in: Serializer.DeserializeState): T
+}
+
+private[scalajs] object Serializer extends SerializerPlatformExtensions {
+  /* Serialization states. Serialization is currently stateless, but in the
+   * future we might want to deduplicate things.
```

Comment:
In the sense that the same value will always get serialized into the same bytes, no matter where it is in the bytestream. It might be a good idea to change this (for example for package prefixes). Since a lot of data seems to be redundant.